crates/sui-graphql-client/src/query_types/mod.rs

```diff
@@ -111,6 +120,19 @@ pub struct MoveObject {
     pub bcs: Option<Base64>,
 }
 
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema = "rpc", graphql_type = "MoveObject")]
+pub struct MoveObjectContents {
+    pub contents: Option<MoveValue>,
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema = "rpc", graphql_type = "MoveValue")]
+pub struct MoveValue {
```

Comment:
I'm not sure it makes sense for the shared `MoveValue` to have both the `BCS` and `JSON` fields -- it seems unlikely that you would query for both in one query, so this would imply some over-fetching.

Reply:
The `DynamicFieldOutput` has a `json` field, which returns the value as a JSON, and the `value` field which returns the BCS and typename. The thought here was that if you know/have the type you could reconstruct it from BCS, but if you only need a field and don't have the type, you use the JSON object.